_episodes/02-file_parsing.md

```diff
@@ -237,48 +252,48 @@ energy = float(words[3])
 >## Exercise on File Parsing (should we move this to the end?)
 Use the provided sapt.out file.  In this output file, the program calculates the interaction energy for an ethene-ethyne complex.  The output reports four interaction energy components: electrostatics, induction, exchange, and dispersion.  Parse each of these energies, in kcal/mole, from the output file.  (Hint: study the file in a text editor to help you decide what to search for.) Calculate the total interaction energy by adding the four components together.  Your code's output should look something like this:
 > ~~~
-> Electrostatics : -2.25850118 kcal/mole
-> Exchange : 2.27730198 kcal/mole
-> Induction : -0.5216933 kcal/mole
-> Dispersion : -0.9446677 kcal/mole
-> Total Energy : 1.4475602000000003 kcal/mole
+> Electrostatics : -2.25850118 kcal/mol
+> Exchange : 2.27730198 kcal/mol
+> Induction : -0.5216933 kcal/mol
+> Dispersion : -0.9446677 kcal/mol
+> Total Energy : 1.4475602000000003 kcal/mol
 > ~~~
 > {: language.python}
 >
 > > ## Solution
 >>
 >> This is one possible solution for the SAPT parsing exercise
 >> ~~~
->> saptout = open('SAPT.out','r')
->> saptlines = saptout.readlines()
->> important_lines=[]
->> energies=[]
->> for line in saptlines:
->>     if 'Electrostatics    ' in line:
->>        electro_line = line
->>        important_lines.append(electro_line)
->>    if 'Exchange       ' in line:
->>        exchange_line = line
->>        important_lines.append(exchange_line)
->>    if 'Induction      ' in line:
->>        induction_line = line
->>        important_lines.append(induction_line)
->>    if 'Dispersion     ' in line:
->>        dispersion_line = line
->>        important_lines.append(dispersion_line)
->>
->> #print(important_lines)
->>
+>> important_lines = []
+>> energies = []
+>> 
+>> with open('SAPT.out','r') as saptout:
+>>     for line in saptout:
+>>         if 'Electrostatics    ' in line:
+>>             electro_line = line
+>>             important_lines.append(electro_line)
+>>         elif 'Exchange       ' in line:
```

Comment:
Use of `elif` here would require us to introduce it at an earlier point in the material. If it is not previously introduced and explained it may be confusing for beginners. I am in favor of this, but we would have to figure out where.

Reply:
I agree, and new users would probably shouldn't need to care about how python run the program (continue running on `if` clause or skip, `else` if clause. Would you like me to switch back to `if` for this PR?

Reply:
I like the idea of showing `elif` and `else`. Let's switch to `if` for this PR, but can you make an issue about introducing `elif` so we can consider where/if to add it?

Thanks for your attention to these lessons!

Reply:
I have created issue #59.